source/Ship.cpp

```diff
@@ -805,12 +796,22 @@ void Ship::Save(DataWriter &out) const
 		}
 		for(const Leak &leak : leaks)
 			out.Write("leak", leak.effect->Name(), leak.openPeriod, leak.closePeriod);
-		for(const auto &it : explosionEffects)
+		
+		using EffectElement = pair<const Effect *const, int>;
+		auto effectSort = [](const EffectElement *lhs, const EffectElement *rhs)
+		{
```

Comment:
The code `if(it.first && it.second)` on line 809/807 suggests that `it.first` might be a null-pointer, but it looks like the check on `it.first` on line 809/807 itself is not actually needed).